Sources/OfflineEntitlements/PurchasedProductsFetcher.swift

```diff
@@ -97,4 +101,48 @@ class PurchasedProductsFetcher: PurchasedProductsFetcherType {
         return result
     }
 
+    private static var hasPendingConsumablePurchase: Bool {
+        get async {
+            return await StoreKit.Transaction.unfinished.contains {
```

Comment:
Just to check, but I'm guessing SK2 will cache the transactions internally right? So this shouldn't take too long. 

Also, will this be executed in a background thread? I'm guessing since the API is using async/await that might be handled internally by SK2 though.

Reply:
The short answer is: it doesn't matter. Concurrency in Swift is different from what you'd expect. Even if this takes 1 hour, it's not stopping the thread, it creates a continuation.